### ISSUE TITLE
Tensor shape or OOM error for smoke3_mov200_f400 training

### ISSUE THREAD

**Issue by Kshitij-Khode** [byungsook/deep-fluids#6]
**TL:DR:** Could you please provide the commands for training a moving 3D smoke source?

**Setup Info**:
Tensorflow version: tensorflow-gpu, 1.12.0, Channel: pypi

**Issue:**
In accordance with run.bat, I generated the smoke3_mov200_f400 dataset using the following command:
```
python main.py --arch=ae --z_num=16 --max_epoch=10 --filter=64 --is_3d=True --lr_max=0.00005 --dataset=smoke3_mov200_f400 --res_x=48 --res_y=72 --res_z=48 --batch_size=4 --num_worker=1
```
Then trained the AE model using,
```
python main.py --arch=ae --z_num=16 --max_epoch=10 --filter=64 --is_3d=True --lr_max=0.00005 --dataset=smoke3_mov200_f400 --res_x=48 --res_y=72 --res_z=48 --batch_size=4 --num_worker=1
```
However, when I try generating latent code set with (trained model stored in log/smoke3_mov200_f400/1213_021319_ae_tag/ )
```
python main.py --is_train=False --load_path=log/smoke3_mov200_f400/1213_021319_ae_tag/  --arch=ae --z_num=16 --max_epoch=20 --is_3d=True --dataset=smoke_mov200_f400 --res_x=48 --res_y=72 --res_z=48 --test_batch_size=4
```

I run into the following error,
```
Traceback (most recent call last):
  File "main.py", line 35, in <module>
    main(config)
  File "main.py", line 21, in main
    trainer = Trainer3(config, batch_manager)
  File "/code/deep-fluids/trainer.py", line 123, in __init__
    self.sess = sv.prepare_or_wait_for_session(config=sess_config)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/training/supervisor.py", line 730, in prepare_or_wait_for_session
    init_feed_dict=self._init_feed_dict, init_fn=self._init_fn)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/training/session_manager.py", line 288, in prepare_session
    config=config)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/training/session_manager.py", line 218, in _restore_checkpoint
    saver.restore(sess, ckpt.model_checkpoint_path)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/training/saver.py", line 1582, in restore
    err, "a mismatch between the current graph and the graph")
tensorflow.python.framework.errors_impl.InvalidArgumentError: Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:
```

I modified the latent code command with --filter=64, but then I ran out of memory,
```
Traceback (most recent call last):
  File "main.py", line 35, in <module>
    main(config)
  File "main.py", line 31, in main
    trainer.test()
  File "/code/deep-fluids/trainer.py", line 308, in test
    self.test_ae()
  File "/code/deep-fluids/trainer.py", line 502, in test_ae
    c = self.sess.run(self.z, {self.x: x})
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 929, in run
    run_metadata_ptr)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1152, in _run
    feed_dict_tensor, options, run_metadata)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1328, in _do_run
    run_metadata)
  File "/root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/python/client/session.py", line 1348, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.ResourceExhaustedError: OOM when allocating tensor with shape[100,64,48,72,48] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
	 [[node AE_1/enc/0_conv/Conv3D (defined at /root/miniconda3/envs/sandbox/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/layers.py:1057)  = Conv3D[T=DT_FLOAT, data_format="NDHWC", dilations=[1, 1, 1, 1, 1], padding="SAME", strides=[1, 1, 1, 1, 1], _device="/job:localhost/replica:0/task:0/device:GPU:0"](_arg_Placeholder_3_0_0/_1283, AE/enc/0_conv/weights/read)]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
```

**Even after reducing batch size via --test_batch_size=1. I still get OOM.**

Since the paper suggests that all the training was done on NVIDIA Titan X (12Gb) GPU and I have the Titan X Pascal (12Gb), I'm assuming that memory shouldn't be the problem.

Could you please provide the commands for training a moving 3D smoke source? run.bat only provided it for smoke3_mov, 2D. Or alternatively please help me identify what I'm doing wrong?

Thanks

**Comment by byungsook** [comment]
You can try the following code:

`python main.py --is_train=False --load_path=AE_MODEL_DIR  --arch=ae --z_num=16 --max_epoch=20 --filter=64 --is_3d=True --dataset=smoke3_mov200_f400 --res_x=48 --res_y=72 --res_z=48 --test_batch_size=5`

Let me know if it works.

**Comment by Kshitij-Khode** [comment]
Sorry. I copy/pasted the wrong command in my initial post. I've updated it with the correct command now.

I used the following command (when dealing with OOM error),
```
python main.py --is_train=False --load_path=log/smoke3_mov200_f400/1213_021319_ae_tag/  --arch=ae --z_num=16 --max_epoch=20 --filter=64 --is_3d=True --dataset=smoke_mov200_f400 --res_x=48 --res_y=72 --res_z=48 --test_batch_size=1
```
Which is similar to what you suggested
```
python main.py --is_train=False --load_path=AE_MODEL_DIR  --arch=ae --z_num=16 --max_epoch=20 --filter=64 --is_3d=True --dataset=smoke3_mov200_f400 --res_x=48 --res_y=72 --res_z=48 --test_batch_size=5
```

I'm still getting OOM error. Did you use --filter=64? Or was a reduced value used in the training? Any other suggestions?

**Comment by Kshitij-Khode** [comment]
I was accidentally limiting the max memory allocation for tensorflow. I removed the limit and I'm now able to generate latent code set.

Thanks for updating run.bat with the relevant commands though. Really appreciate it.
